tribe-customizations/v2/week/grid-body/events-day/event.php

```diff
@@ -0,0 +1,128 @@
+ *
+ * @version 5.0.0
+ *
+ * @var WP_Post $event The event post object with properties added by the `tribe_get_event` function.
+ *
+ * @see tribe_get_event() For the format of the event object.
+ *
+ */
+
+use Tribe__Date_Utils as Dates;
+
+$classes = [ 'tribe-events-pro-week-grid__event' ];
+
+if ( ! empty( $event->is_past ) ) {
+	$classes[] = 'tribe-events-pro-week-grid__event--past';
+}
+
+if ( ! empty( $event->featured ) ) {
+	$classes[] = 'tribe-events-pro-week-grid__event--featured';
+}
+
+/*
+ * Some CSS classes (i.e. vertical position, duration and sequence) have been calculated in the Week View.
+ * Here we add them to the ones that should be applied to the event element.
+ */
+$classes = array_merge( array_values( $classes ), array_values( $event->classes ) );
+$classes = get_post_class( $classes, $event->ID );
+$data_js = [ 'tribe-events-pro-week-grid-event-link', 'tribe-events-tooltip' ];
+
+/**
+ * Get start time in seconds
+ */
+$start_time = Dates::time_between( $event->dates->start->format( 'Y-m-d 0:0:0' ),
+                                   $event->dates->start->format( Dates::DBDATETIMEFORMAT ) );
+
+/**
+ * Get the extension settings
+ */
+$grid_start_time = $ext_options['grid_start_time'];
+$grid_end_time   = $ext_options['grid_end_time'];
+
+$found_start_time = false;
+
+$pattern = '/(tribe-events-pro-week-grid__event--t-)/';
+
+foreach ( $classes as $key => $class ) {
+	// Check if event has the vertical offset class.
+	// An event starting at 12am doesn't have a vertical offset class.
+	if ( preg_match( $pattern, $class ) ) {
+		$found_start_time = true;
+	}
+}
+
+// Hide the ones that would be in the 12-1 row. (They don't have the class.)
+if ( ! $found_start_time && $grid_start_time > 0 ) {
+	$classes[] = 'tribe-common-a11y-visual-hide';
+} else {
+	// Set the new class
+	foreach ( $classes as $key => $class ) {
+		$vertical_offset_class = strpos( $class, 'tribe-events-pro-week-grid__event--t-' );
+
+		if ( preg_match( $pattern, $class ) ) {
+			// Remove the old vertical positioning
+			unset( $classes[ $key ] );
+
+			// Grab the vertical offset
+			$event_start_time = str_replace( 'tribe-events-pro-week-grid__event--t-', '', $class );
+			$time_split       = explode( '-', $event_start_time );
+
+			// Set the new starting hour / offset of the event
+			$new_event_start_hour = (int) $time_split[0] - (int) $grid_start_time;
+
+			// Hide if...
+			if ( // Time is off the chart (negative start time)
+				( $new_event_start_hour <= 0 ) ||
+				// Time is in the first 15 minutes of the grid
+				( $grid_start_time === $new_event_start_hour && $time_split[1] <= 15 ) ||
+				// Time is before the grid start time
+				( $new_event_start_hour < $grid_start_time ) ||
+				// Time + 1 hour is after the grid end time (for long events at the end of the day
+				( $grid_end_time <= $time_split[0] + (int) $grid_start_time + 1 ) ) {
+				$classes[ $key ] = 'tribe-common-a11y-visual-hide';
+				break;
+			}
```

Comment:
- initial comment should be on new line
- closing of IF should be on own line
- Put ORs `||` at start, not end (see "Control structures" at https://moderntribe.github.io/products-engineering/docs/code-standards/php/)